pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -76,23 +76,6 @@ func (e *aksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 		return cluster, nil
 	}
 
-	if err := e.deployAKSOperator(); err != nil {
-		failedToDeployAKSOperatorErr := "failed to deploy aks-operator: %v"
-		var conditionErr error
-		if cluster.Spec.AKSConfig.Imported {
-			cluster, conditionErr = e.SetFalse(cluster, apimgmtv3.ClusterConditionPending, fmt.Sprintf(failedToDeployAKSOperatorErr, err))
-			if conditionErr != nil {
-				return cluster, conditionErr
-			}
-		} else {
-			cluster, conditionErr = e.SetFalse(cluster, apimgmtv3.ClusterConditionProvisioned, fmt.Sprintf(failedToDeployAKSOperatorErr, err))
-			if conditionErr != nil {
-				return cluster, conditionErr
-			}
-		}
-		return cluster, err
-	}
-
```

Comment:
Nit: Should we handle the scenario where the CRD isn't installed yet more gracefully or wait to proceed until it is installed? Otherwise, errors are likely to occur on L87. We could also return an error saying "waiting for CRD to be created.

Reply:
Since the chart operator is based on `v3.cluster`. Once chart is installed it will change v3.cluster which trigger this controller, so it should eventually go through? But yeah I can check if the crd exists and give a more graceful error.